File: Samples/DetermineCorruptPage/C#/DetermineCorruptPage.cs
using System.Windows.Forms;

using BitMiracle.LibTiff.Classic;

namespace BitMiracle.LibTiff.Samples
{
    public static class DetermineCorruptPage
    {
        public static void Main()
        {
            using (Tiff image = Tiff.Open(@"Sample Data\127.tif", "r"))
            {
                if (image == null)
                {
                    MessageBox.Show("Could not load incoming image");
                    return;
                }

                int numberOfDirectories = image.NumberOfDirectories();
                for (int i = 0; i < numberOfDirectories; ++i)
                {
                    image.SetDirectory((short)i);

                    int width = image.GetField(TiffTag.IMAGEWIDTH)[0].ToInt();
                    int height = image.GetField(TiffTag.IMAGELENGTH)[0].ToInt();

                    int imageSize = height * width;
                    int[] raster = new int[imageSize];

                    if (!image.ReadRGBAImage(width, height, raster, true))
                    {
                        MessageBox.Show("Page " + i + " is corrupted");
                        return;
                    }
                }
            }
        }
    }
}

File: Samples/DetermineCorruptPage/VB.NET/DetermineCorruptPage.vb
Imports System.Windows.Forms

Imports BitMiracle.LibTiff.Classic

Namespace BitMiracle.LibTiff.Samples
    Public NotInheritable Class DetermineCorruptPage
        Private Sub New()
        End Sub
        Public Shared Sub Main()
            Using image As Tiff = Tiff.Open("Sample Data\127.tif", "r")
                If image Is Nothing Then
                    MessageBox.Show("Could not load incoming image")
                    Return
                End If

                Dim numberOfDirectories As Integer = image.NumberOfDirectories()
                For i As Integer = 0 To numberOfDirectories - 1
                    image.SetDirectory(CShort(i))

                    Dim width As Integer = image.GetField(TiffTag.IMAGEWIDTH)(0).ToInt()
                    Dim height As Integer = image.GetField(TiffTag.IMAGELENGTH)(0).ToInt()

                    Dim imageSize As Integer = height * width
                    Dim raster As Integer() = New Integer(imageSize) {}

                    If Not image.ReadRGBAImage(width, height, raster, True) Then
                        MessageBox.Show("Page " + i.ToString() + " is corrupted")
                        Return
                    End If
                Next
            End Using
        End Sub
    End Class
End Namespace